web/packages/teleterm/src/services/config/createAppConfigSchema.ts

```diff
@@ -0,0 +1,189 @@
+import { z } from 'zod';
+
+import { Platform } from 'teleterm/mainProcess/types';
+
+import { getKeyboardShortcutSchema } from './getKeyboardShortcutSchema';
+
+export type AppConfig = z.infer<ReturnType<typeof createAppConfigSchema>>;
+
+export const createAppConfigSchema = (platform: Platform) => {
```

Comment:
Off topic nit but lately I've been reconsidering if putting the verbs in the names of the files themselves is a good idea. I feel like inside this dir we could have `appConfigSchema.ts` and `keyboardShortcutSchema.ts` and it'd be totally fine. We also don't have that many files starting with get/create in the project.

It'd also help clean up the naming between `createAppConfigSchema` and `getKeyboardShortcutSchema`, at least on the file name level. 😏 I think I prefer the verb "create" or "make" for this scenario over "get" as what we're really doing here is instantiating something new.

Reply:
I renamed `getKeyboardShortcutSchema` too :) 

I don't have a strong opinion on the file names, and even in a `services/config/` module we were inconsistent, for example a function `createConfigService` was exported from `configService.ts` file.

Sometimes naming the file the same as a main/only export makes sense for me, for example in `tshd/createClient.ts` (maybe I'm just used to it). Would you see it as `tshd/client.ts`?

Reply:
> Would you see it as `tshd/client.ts`?

I suppose so? I don't have a hard opinion on this atm, but overall I'm thinking about shifting focus when naming from particular functions to overall concepts.

If JS was more opinionated about imports and files then it'd all be easier. For example each file in Elm is a separate module with a specific namespace, so you could have `ConfigService.elm` and you'd expect to find a function called `ConfigService.init`. Inside `ConfigService.elm`, you can use `init` without the namespace qualifier. Outside of that module, you typically import the whole module and call `ConfigService.init` but you can also import a specific function without the namespace qualifier or use an alias for the namespace.

In JS there's no notion of this namespace which makes things harder.

---

In general I think I'd aim at reducing the number of files we create, each function doesn't have to live in a separate file. Take a look at `e/web/teleport/src/services/recovery/makeRecoveryToken.ts`, this function could as well live in `e/web/teleport/src/services/recovery/recovery.ts`.

Bigger functions such as `web/packages/teleport/src/services/recordings/makeRecording.ts` could live in `recording.ts` as well, especially since they define functions that are related to a recording but might be useful outside of the context of `makeRecording`, for example `formatDuration`.

This all seems easy and clear as long as we we're talking about non-React code. As soon as we enter React then we potentially have a conflict between `recovery.ts` and a file for a component called `Recovery`.